index.js

```diff
@@ -20,12 +26,23 @@ module.exports = async function percySnapshot(t, name, options) {
 
     // Serialize and capture the DOM
     /* istanbul ignore next: no instrumenting injected code */
-    let { domSnapshot, url } = await t.eval(() => ({
+    let { domSnapshot, url, proxyUrl } = await t.eval(() => ({
       /* eslint-disable-next-line no-undef */
       domSnapshot: PercyDOM.serialize(options),
-      url: document.URL
+      // window.location.href as well as document.URL is overriden to return correct test page url
+      // and does not include proxy url
+      url: window.location.href || document.URL,
+      // We get proxy URL from internal implementation of hammerhead, there is no API to get it
+      // otherwise in testcafe. We need this because when https sites are being tested with http
+      // proxy, we are unable to do resource discovery due to SSL errors [ testcafe replaces
+      // all urls with proxied urls and causes http calls from https page in asset discovery ]
+      proxyUrl: window['%hammerhead%']?.utils?.url?.getProxyUrl('')
```

Comment:
```suggestion
      proxyUrl: window['%hammerhead%']?.utils?.url?.getProxyUrl('')
```

do we want to fallback this if the API doesn't contain this implementation?

Reply:
There is no fallback as of now, so if this is `undefined` then we skip updating `url` with proxy and use original url instead [ which is existing behaviour ] check if condition below.